### PR TITLE
New Feature: Documentation for React and DOM properties

### PR DESCRIPTION
**Pull request by jlaluces123**
This PR Includes:

- A new section on the documentation page of each component detailing information on implicit DOM/React properties that are available
- New DomDocs component that contains the section text
- Resolves grommet/grommet#4969

### CODE REVIEW

File: src/components/Doc/DomDocs.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      </Paragraph>
+      <ReactDocs />
+    </Box>
+  );
+};
+
+const ReactDocs = () => {
+  return (
+    <Paragraph>
+      Feel free to read on the{' '}
```
Comment (ShimiSun) [inline src/components/Doc/DomDocs.js:50]:
```suggestion
      Also, feel free to read on the{' '}
```

File: src/components/Doc/DomDocs.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        href="https://reactjs.org/docs/handling-events.html"
+        label="types of React events available"
+        target="_blank"
+      />
+      , or see{' '}
+      <Anchor
+        href="https://reactjs.org/docs/dom-elements.html"
+        label="how DOM attributes change in React."
+        target="_blank"
+      />
```
Comment (ShimiSun) [inline src/components/Doc/DomDocs.js:61]: I would think these props should include React `as` prop documentation.

**Review by ShimiSun — COMMENTED** [review]
Fantastic! I'm so glad you are working on this issue and helping us improve the documentation of grommet. Great progress and discussions.

File: src/components/Doc/DomDocs.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        margin={{ top: 'medium', bottom: 'xlarge' }}
+      >
+        <Heading level={2} margin="none">
+          React/DOM Properties
+        </Heading>
+        <Anchor href="#dom-react" icon={<LinkIcon color="light-4" />} />
+      </Box>
+      <Paragraph>
+        At its core, the {name} component is a regular {`<${intrinsicElement}>`}{' '}
+        element. Thus, both the DOM&apos;s and React&apos;s properties, methods,
```
Comment (taysea) [inline src/components/Doc/DomDocs.js:26]:
```suggestion
        element. Thus, both DOM and React properties, methods,
```

Comment (taysea) [reply src/components/Doc/DomDocs.js:26]: small wording suggestion

File: src/components/Doc/DomDocs.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        />
+      </Paragraph>
+      <ReactDocs />
+    </Box>
+  );
+};
+
+const ReactDocs = () => (
+    <Paragraph>
+      Also, feel free to read on the{' '}
```
Comment (taysea) [inline src/components/Doc/DomDocs.js:49]:
```suggestion
      Also, feel free to read about the{' '}
```

File: src/components/Doc/DomDocs.js (hunk 2 above)
Comment (jlaluces123) [reply src/components/Doc/DomDocs.js:60]: Just to clarify, are we referring to the `as` import alias of `import {component as componentName} from ...` or Grommet's `as` prop (ex. `Box as = "button">...`)?

File: src/components/Doc/DomDocs.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        out this{' '}
+        <Anchor
+          href={`https://developer.mozilla.org/en-US/docs/Web/HTML/Element/${intrinsicElement}`}
+          label="MDN Web Documents page."
+          target="_blank"
+        />{' '}
+        To learn more about DOM events and methods, you can read more{' '}
+        <Anchor
+          href="https://developer.mozilla.org/en-US/docs/Web/Events"
+          label="here."
```
Comment (taysea) [inline src/components/Doc/DomDocs.js:38]:
```suggestion
          label="MDN Web Documents Events page"
```
I think you might've had this on a previous iteration, but it's better for screen readers if the Anchor text is descriptive about where it's going. I'd recommend rephrasing to something more focused.

File: src/components/Doc/DomDocs.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        label="types of React events available"
+        target="_blank"
+      />
+      , or see{' '}
+      <Anchor
+        href="https://reactjs.org/docs/dom-elements.html"
+        label="how DOM attributes change in React."
+        target="_blank"
+      />
+      {' '}Working in tandem with Styled Components, you also have access to the {' '}
```
[the remark this answers is not in the document]
Comment (ShimiSun) [reply src/components/Doc/DomDocs.js:61]: Thank you for double-checking, might be a cache issue. Let me take a look.

[the remark this answers is not in the document]
Comment (taysea) [reply src/components/Doc/DomDocs.js:61]: I am able to see it correctly in my browser!

File: src/components/Doc/Doc.js [hunk 7]
```diff
@@ -10,6 +10,7 @@ import RoutedButton from '../RoutedButton';
 import { Code } from './Code';
 import { Props } from './Props';
 import { ThemeProps } from './ThemeProps';
+import { DomDocs } from './DomDocs';
```
Comment (ShimiSun) [inline src/components/Doc/Doc.js:13]:
```suggestion
import { DomProps } from './DomDocs';
```

For consistency.

**Review by ShimiSun — APPROVED** [review]

**Pull request merged**